app/src/main/java/de/danoeh/antennapod/fragment/EpisodesListFragment.java

```diff
@@ -457,7 +462,11 @@ protected void updateToolbar() {
     @Subscribe(sticky = true, threadMode = ThreadMode.MAIN)
     public void onEventMainThread(FeedUpdateRunningEvent event) {
         if (toolbar.getMenu().findItem(R.id.refresh_item) != null) {
-            MenuItemUtils.updateRefreshMenuItem(toolbar.getMenu(), R.id.refresh_item, event.isFeedUpdateRunning);
+            if (event.isFeedUpdateRunning) {
+                swipeRefreshLayout.setRefreshing(true);
+                new Handler(Looper.getMainLooper()).postDelayed(() -> swipeRefreshLayout.setRefreshing(false),
+                        getResources().getInteger(R.integer.swipe_to_refresh_duration_in_ms));
+            }
```

Comment:
Because the swipe layout is now used as an indicator, it should stay visible during the entire refresh instead of just `swipe_to_refresh_duration_in_ms`

```suggestion
            swipeRefreshLayout.setRefreshing(event.isFeedUpdateRunning);
```